llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -1165,6 +1204,46 @@ AliasResult BasicAAResult::aliasGEP(
   if (!V1Size.hasValue() || !V2Size.hasValue())
     return AliasResult::MayAlias;
 
+  // VScale Alias Analysis
```

Comment:
Please drop this code from the initial patch -- let's make sure everything works with vscale in the decomposition first, before we extend to also handling scalable location size.

Reply:
I was hoping that more of the existing code could handle vscales already if they were treated like variables, but I think there might need to be something that know that vscale in the offset is the same as the vscale in the typesize.